odl/tomo/operators/ray_trafo.py

```diff
@@ -146,6 +149,21 @@ def __init__(self, reco_space, geometry, variant, **kwargs):
             if geometry.ndim > 2 and impl.endswith('cpu'):
                 raise ValueError('`impl` {!r}, only works for 2d'
                                  ''.format(impl_in))
+
+            # Filter axis-alinged initial detector position in parallel 3d
+            # geometry since it's broken for some ASTRA versions
+            if isinstance(geometry, Parallel3dAxisGeometry):
+                axis = geometry.axis
+                init_vec = (geometry._det_init_pos /
+                            np.linalg.norm(geometry._det_init_pos))
+                # Almost the same
+                if (np.isclose(np.dot(axis, init_vec), 1) and
+                        not astra_supports('par3d_axis_aligned_det_init_pos')):
```

Comment:
Okay, I just did that. The exact error scenario is that `geometry.det_to_src(angle, mid_pt)` in [this line](https://github.com/odlgroup/odl/blob/4d886be4cf761eb297e257b5487abc7634ba5754/odl/tomo/backends/astra_setup.py#L304) has 0 component along `geometry.axis`, i.e., when the two are perpendicular. In that case, [this line](https://github.com/astra-toolbox/astra-toolbox/blob/86ed34e9a5fa408c9338cd5c2871f7f7953806b7/src/CompositeGeometryManager.cpp#L228) in ASTRA, and in particular [this division](https://github.com/astra-toolbox/astra-toolbox/blob/71db6331f8dd0d5abbeee92977af01293be4f427/src/GeometryUtil3D.cpp#L171) in `backprojectPointX` yields +-infinity or NaN(?) (the TODO above the function is telling :-)). The `fRayX` member refers to the [first column of the vectors in the vec geometries](https://github.com/astra-toolbox/astra-toolbox/blob/71db6331f8dd0d5abbeee92977af01293be4f427/src/ParallelVecProjectionGeometry3D.cpp#L105), which I checked will be zero in the error case.